src/Symfony/Component/DependencyInjection/Compiler/ValidateEnvPlaceholdersPass.php

```diff
@@ -51,12 +51,22 @@ public function process(ContainerBuilder $container)
         foreach ($resolvingBag->getEnvPlaceholders() + $resolvingBag->getUnusedEnvPlaceholders() as $env => $placeholders) {
             $values = [];
             if (false === $i = strpos($env, ':')) {
-                $default = $defaultBag->has("env($env)") ? $defaultBag->get("env($env)") : self::TYPE_FIXTURES['string'];
-                $defaultType = null !== $default ? get_debug_type($default) : 'string';
+                [$default, $defaultType] = $this->getParameterDefaultAndDefaultType("env($env)", $defaultBag);
                 $values[$defaultType] = $default;
             } else {
                 $prefix = substr($env, 0, $i);
-                foreach ($envTypes[$prefix] ?? ['string'] as $type) {
+
+                if ('default' === $prefix) {
+                    [$parameter, $envVar] = explode(':', substr($env, $i + 1));
+                    [$defaultParameter, $defaultParameterType] = $this->getParameterDefaultAndDefaultType($parameter, $defaultBag);
+                    [$defaultEnv, $defaultEnvType] = $this->getParameterDefaultAndDefaultType("env($envVar)", $defaultBag);
+
+                    $types = array_unique([$defaultParameterType, $defaultEnvType]);
```

Comment:
```suggestion
                    $types = array_unique([$defaultParameterType, $defaultEnvType]) ?: ['string'];
```
So that we haven't an empty array